src/dmd/root/dcompat.h

```diff
@@ -43,6 +43,9 @@ typedef unsigned d_size_t;
         __APPLE__ && __SIZEOF_SIZE_T__ == 8
 // DMD versions between 2.079 and 2.081 mapped D ulong to uint64_t on OS X.
 typedef uint64_t d_size_t;
+#elif defined(__OpenBSD__) && !defined(__LP64__)
+// size_t is 'unsigned long', which makes it mangle differently than D's 'uint'
+typedef unsigned d_size_t;
```

Comment:
Since you are making changes here: I think the `length` field of `DArray` should also be `d_size_t`.

Reply:
Force pushed this.

Reply:
Well, that blew up immediately. I'll revisit this immediately after this goes in, since it may be a little more invasive than a simple replacement.